src/Open3D/Visualization/Visualizer/GuiVisualizer.cpp

```diff
@@ -236,6 +286,127 @@ GuiVisualizer::GuiVisualizer(
+            }
+        });
+        ShowDialog(dlg);
+    });
+    loadButtons->AddChild(lightSettings.wgtLoadAmbient);
+
+    lightSettings.wgtLoadSky = std::make_shared<gui::Button>("Load skybox");
+    lightSettings.wgtLoadSky->SetOnClicked([this, renderScene]() {
+        auto dlg = std::make_shared<gui::FileDialog>(
+                gui::FileDialog::Type::OPEN, "Open skybox", GetTheme());
+        dlg->AddFilter(".ktx", "Khronos Texture (.ktx)");
+        dlg->SetOnCancel([this]() { this->CloseDialog(); });
+        dlg->SetOnDone([this, renderScene](const char *path) {
+            this->CloseDialog();
+            auto newSky = GetRenderer().AddSkybox(ResourceLoadRequest(path));
+            if (newSky) {
+                impl_->lightSettings.hSky = newSky;
+
+                renderScene->SetSkybox(newSky);
+            }
+        });
+        ShowDialog(dlg);
+    });
+    loadButtons->AddChild(lightSettings.wgtLoadSky);
+    lightSettings.wgtBase->AddChild(loadButtons);
+
+    lightSettings.wgtBase->AddChild(gui::Horiz::MakeFixed(8));
+    lightSettings.wgtBase->AddChild(
+            std::make_shared<gui::Label>("Light switches:"));
+    auto checkboxes = std::make_shared<gui::Horiz>(0, gui::Margins{0, 0, 0, 8});
+    lightSettings.wgtAmbientEnabled =
+            std::make_shared<gui::Checkbox>("Ambient");
+    lightSettings.wgtAmbientEnabled->SetChecked(true);
+    lightSettings.wgtAmbientEnabled->SetOnChecked(
+            [this, renderScene](bool checked) {
+                if (checked) {
+                    renderScene->SetIndirectLight(impl_->lightSettings.hIbl);
+                } else {
+                    renderScene->SetIndirectLight(IndirectLightHandle());
+                }
+            });
+    checkboxes->AddChild(lightSettings.wgtAmbientEnabled);
+    lightSettings.wgtSkyEnabled = std::make_shared<gui::Checkbox>("Sky");
+    lightSettings.wgtSkyEnabled->SetChecked(true);
+    lightSettings.wgtSkyEnabled->SetOnChecked(
+            [this, renderScene](bool checked) {
+                if (checked) {
+                    renderScene->SetSkybox(impl_->lightSettings.hSky);
+                } else {
+                    renderScene->SetSkybox(SkyboxHandle());
+                }
+            });
+    checkboxes->AddChild(lightSettings.wgtSkyEnabled);
+    lightSettings.wgtDirectionalEnabled =
+            std::make_shared<gui::Checkbox>("Sun");
+    lightSettings.wgtDirectionalEnabled->SetChecked(true);
+    lightSettings.wgtDirectionalEnabled->SetOnChecked(
+            [this, renderScene](bool checked) {
+                renderScene->SetEntityEnabled(
+                        impl_->lightSettings.hDirectionalLight, checked);
+            });
+    checkboxes->AddChild(lightSettings.wgtDirectionalEnabled);
+    lightSettings.wgtBase->AddChild(checkboxes);
+
+    lightSettings.wgtIntensity = AddAttributeSlider(
+            gui::Slider::INT, 0.0, 1000000.0, lightDescription.intensity,
```

Comment:
I think the max should be something more like 150000 instead of 1e6. Anything over about 100000 (which I think is the Filament suggested sunlight value) is too bright, which then makes the useful ranges crammed into a couple of pixels.